lib/cipher.rb

```diff
@@ -33,4 +33,52 @@ def shift(char, shift)
       char
     end
   end
+
+  def crack_key(message)
+    shift_labels = %w(0 1 2 3).rotate(message.length % 4)
+    shifts = []
+    shifts << ALPHA.index(message[-4]) - ALPHA.index(" ")
+    shifts << ALPHA.index(message[-3]) - ALPHA.index("e")
+    shifts << ALPHA.index(message[-2]) - ALPHA.index("n")
+    shifts << ALPHA.index(message[-1]) - ALPHA.index("d")
+
+    shift_labels.zip(shifts).to_h
+    # return is a hash w shift values to decrypt, using keys 0-3 as a-d
+  end
+
+  def crack_chars(message)
```

Comment:
This lets me decode the _ciphertext_ with the shift hash from above method `#crack_key`